mailparse.c

```diff
@@ -172,57 +178,52 @@ PHP_MSHUTDOWN_FUNCTION(mailparse)
 
 static inline php_mimepart *mimemsg_get_object(zval *object TSRMLS_DC)
 {
-	zval **zpart;
+	zval *zpart;
 	php_mimepart *part;
-	int type;
-	
 
 	if (Z_TYPE_P(object) != IS_OBJECT) {
 		return NULL;
 	}
 
-	if (zend_hash_index_find(Z_OBJPROP_P(object), 0, (void**)&zpart) == FAILURE) {
+	if ((zpart = zend_hash_index_find(Z_OBJPROP_P(object), 0)) == NULL) {
 		return NULL;
 	}
 
-	part = zend_list_find(Z_LVAL_PP(zpart), &type);
-
-	if (type != le_mime_part)
+	if ((mailparse_fetch_mimepart_resource(part, zpart)) == NULL) {
 		return NULL;
+	}
 
 	return part;
 }
 
 static int mailparse_mimemessage_populate(php_mimepart *part, zval *object TSRMLS_DC)
 {
-	zval *tmp;
+	zval tmp;
 
-	MAKE_STD_ZVAL(tmp);
-	mailparse_get_part_data(part, tmp TSRMLS_CC);
-	add_property_zval(object, "data", tmp);
-	Z_DELREF_P(tmp);
+	mailparse_get_part_data(part, &tmp TSRMLS_CC);
+	add_property_zval(object, "data", &tmp);
+	Z_DELREF_P(&tmp);
```

Comment:
maybe use zval_ptr_dtor is better, if the type is not REFCOUNTED, this may lead to segfault.